lib/walker.ts

```diff
@@ -450,6 +451,21 @@ class Walker {
     assert(typeof task.file === 'string');
     const realFile = toNormalizedRealPath(task.file);
 
+    const ignore = task.marker?.toplevelConfig.pkg?.ignore;
+    if (ignore) {
+      // check if the file matches one of the ignore regex patterns
+      const match = ignore.some((pattern) =>
+        new RegExp(pattern).test(realFile),
+      );
+
+      if (match) {
+        log.debug(
+          `Ignoring file: ${realFile} due to top level config ignore pattern`,
+        );
+        return;
+      }
```

Comment:
I would use `minimatch` here to test files using globs not regex

Reply:
sure thing!

Reply:
Done, let me know if you have opinions on the plugin options, i kept the default ones.